Sources/Basics/Observability.swift

```diff
@@ -571,53 +556,6 @@ extension Diagnostic {
     }
 }
 
-@available(*, deprecated, message: "temporary for transition DiagnosticsEngine -> DiagnosticsEmitter")
-extension ObservabilitySystem {
-    public convenience init(diagnosticEngine: DiagnosticsEngine) {
-        self.init(DiagnosticsEngineAdapter(diagnosticEngine: diagnosticEngine))
-    }
-
-    private struct DiagnosticsEngineAdapter: ObservabilityHandlerProvider, DiagnosticsHandler {
-        let diagnosticEngine: DiagnosticsEngine
-
-        var diagnosticsHandler: DiagnosticsHandler { self }
-
-        func handleDiagnostic(scope: ObservabilityScope, diagnostic: Diagnostic) {
-            diagnosticEngine.emit(.init(diagnostic))
-        }
-    }
-}
-
-@available(*, deprecated, message: "temporary for transition DiagnosticsEngine -> DiagnosticsEmitter")
-extension TSCBasic.Diagnostic {
-    public init(_ diagnostic: Diagnostic) {
-        let location: DiagnosticLocation
-        if let legacyLocation = diagnostic.metadata?.legacyDiagnosticLocation {
-            location = legacyLocation.underlying
-        } else {
-            location = UnknownLocation.location
-        }
-
-        let data: DiagnosticData
-        if let legacyData = diagnostic.metadata?.legacyDiagnosticData {
-            data = legacyData.underlying
-        } else {
-            data = StringDiagnostic(diagnostic.message)
-        }
-
-        switch diagnostic.severity {
-        case .error:
-            self = .init(message: .error(data), location: location)
-        case .warning:
-            self = .init(message: .warning(data), location: location)
-        case .info:
-            self = .init(message: .note(data), location: location)
-        case .debug:
-            self = .init(message: .note(data), location: location)
-        }
-    }
-}
-
 @available(*, deprecated, message: "temporary for transition DiagnosticsEngine -> DiagnosticsEmitter")
```

Comment:
need this too?

Reply:
all the `legacyXXX` ObservabilityMetadata need to go if we remove the above basically

Reply:
I think I basically paired this down to "what do we need to still build/pass the tests", but I can take another look.

Reply:
We have convinced ourselves that this seems fine, so this should be ready to merge.